openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIComponentMapper.java

```diff
@@ -33,63 +34,135 @@
+        return diagnostics;
+    }
     /**
      * This function for doing the mapping with ballerina record to object schema.
      *
      * @param schema         Map of current schemas
      * @param typeSymbol     Record Name as a TypeSymbol
      */
     public void createComponentSchema(Map<String, Schema> schema, TypeSymbol typeSymbol) {
+        if (schema == null) {
+            schema = new HashMap<>();
+        }
         // Getting main record description
         String componentName = ConverterCommonUtils.unescapeIdentifier(typeSymbol.getName().orElseThrow().trim());
         Map<String, String> apiDocs = getRecordFieldsAPIDocsMap((TypeReferenceTypeSymbol) typeSymbol, componentName);
+        String typeDoc = null;
+        if (apiDocs.size() > 0) {
+            typeDoc = apiDocs.get(typeSymbol.getName().get());
+        }
         TypeReferenceTypeSymbol typeRef = (TypeReferenceTypeSymbol) typeSymbol;
+        TypeSymbol type = typeRef.typeDescriptor();
         // Handle record type request body
-        if (typeRef.typeDescriptor() instanceof RecordTypeSymbol) {
-            // Handle typeInclusions with allOf type binding
-            handleRecordTypeSymbol((RecordTypeSymbol) typeRef.typeDescriptor(), schema, componentName, apiDocs);
-        } else if (typeRef.typeDescriptor() instanceof IntersectionTypeSymbol) {
-            List<TypeSymbol> typeSymbols = ((IntersectionTypeSymbol) typeRef.typeDescriptor()).memberTypeDescriptors();
-            for (TypeSymbol symbol: typeSymbols) {
-                if (!(symbol instanceof RecordTypeSymbol)) {
-                    continue;
+        switch (type.typeKind()) {
+            case RECORD:
+                // Handle typeInclusions with allOf type binding
+                handleRecordTypeSymbol((RecordTypeSymbol) type, schema, componentName, apiDocs);
+                break;
+            case INTERSECTION:
+                List<TypeSymbol> typeSymbols = ((IntersectionTypeSymbol) type).memberTypeDescriptors();
+                for (TypeSymbol symbol: typeSymbols) {
+                    if (!(symbol instanceof RecordTypeSymbol)) {
+                        continue;
+                    }
+                    handleRecordTypeSymbol((RecordTypeSymbol) symbol, schema, componentName, apiDocs);
                 }
-                handleRecordTypeSymbol((RecordTypeSymbol) symbol, schema, componentName, apiDocs);
-            }
+                break;
+            case STRING:
+                schema.put(componentName, new StringSchema().description(typeDoc));
+                components.setSchemas(schema);
+                break;
+            case INT:
+                schema.put(componentName, new IntegerSchema().description(typeDoc));
+                components.setSchemas(schema);
+                break;
+            case DECIMAL:
+                Objects.requireNonNull(schema.put(componentName, new NumberSchema().format(DOUBLE)))
```

Comment:
why are we having this ` Objects.requireNonNull()` usage only within this `case` clause? And shouldn't the description be set to the `NumberSchema` like in other cases?

Reply:
Yeah, we can add within NumberSchema, mistakenly chosen outbound. Will fix this